keyring/keyring.go

```diff
@@ -0,0 +1,110 @@
+package keyring
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"encoding/base64"
+	"os"
+	"path/filepath"
+
+	"github.com/deislabs/go-bindle/types"
+	"github.com/pelletier/go-toml"
+)
+
+// GenerateSigningKey generates a keypair for signing Bindle invoices
+// The return types are the public key (wrapped in a SignatureKey), the private key, and any error
+func GenerateSignatureKey(author, role string) (*types.SignatureKey, []byte, error) {
```

Comment:
It would probably be good to validate the given role to make sure it is one of the supported roles. Right now someone could create and use a key that has an unsupported role